### DISCUSSION TITLE
SubCategories and Rest

### DISCUSSION THREAD

**Discussion by agrufu** [trizen/obmenu-generator#53] [Ideas]
I love your application to generate menus in Openbox.
I'm not an expert programmer, and I don't understand much Perl, but I'd like to modify obmenu-generator so that it generates SubCategories inside Categories, and this applications that aren't inside any subcategory will be inside the category menu.
For example:
System -> File Manager -> Pcmanfm ...
                     Virtualize -> Virtualbox...
                     Bleachbit (out of any submenu)

It could be generated taking as reference the categories indicated in the .descktop file, but prioritizing as a type of priority.
For example:
...
Categories=GNOME;System;Filesystem;
...
In this case the first category would be GNOME, within this category would be System and within it File System.
And if there was only the GNOME category, then the application would appear within this category, below the subcategories.
I don't know if I explained well.

**Comment by MatanzaBR** [comment]
Para adicionar uma nova categoria

To add a new category

Exemplo:


$ cat ~/.config/obmenu-generator/schema.pl
```

{cat => ['Sistema',      'Manual',      'applications-system']},

```

$ cat ~/.local/share/applications/"busca pacotes instalados.desktop" | grep Categories

```
Categories=Sistema;
```

Resultado:

![nova-categoria](https://github.com/user-attachments/assets/d6364cd2-2823-4bb0-a3f3-97d9c1340509)
